github-runner-manager/src/github_runner_manager/reactive/runner_manager.py

```diff
@@ -128,3 +128,8 @@ def reconcile(
     )
 
     return ReconcileResult(processes_diff=processes_created, metric_stats=metric_stats)
+
+
+def flush_reactive_processes() -> None:
+    """Flush all the reactive processes."""
+    process_manager.kill_reactive_processes()
```

Comment:
I'm wondering if you want to make a classes for this pipeline and inject the dependencies, 
i.e. ReactiveManager(process_manager: ProcessManager)

The reason is because during testing, monkeypatching receives a string of things to patch and cannot be refactored easily with the help of IDE's intelliJ. Also, having the clear dependencies of classes will help us see the domain components easily - to help see the workflow clearer.

Reply:
- I feel using this style only due to testing, would be a large change. This would need to be done at a team-level. 
- Under the current method the using mocking would be `monkeypatch.setattr("github_runner_manager.reactive.runner_manager.kill_reactive_processes", mock_func)`. I think this not worse that the functional programming injection, since both mock at the function-level.

I think we need to bring this up in team discussion. I have added this to agenda.